### PR TITLE
Added checks if kubernetes_node group is defined

### PR DESCRIPTION
**Pull request by to-bar**
- Added checks if kubernetes_node group is defined (#750)
- Do not untaint master in upgrade role

### CODE REVIEW

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml [hunk 1]
```diff
@@ -60,7 +60,8 @@
     KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
   shell: kubectl drain {{ inventory_hostname }} --ignore-daemonsets --delete-local-data
   when:
-    - groups['kubernetes_node'] is defined and groups['kubernetes_node']|list|length > 0 # drain master only if there is at least one worker node
+    - groups['kubernetes_node'] is defined
+    - groups['kubernetes_node']|length > 0 # drain master only if there is at least one worker node
```
Comment (toszo) [inline core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml]:
```suggestion
    - groups['kubernetes_node']|list|length > 0 # drain master only if there is at least one worker node
```

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml [hunk 2]
```diff
@@ -123,29 +124,9 @@
...
-    - (groups['kubernetes_master']|list|length == 1) and (groups['kubernetes_node'] is defined and groups['kubernetes_node']|list|length == 0)
-  changed_when: false
-
-- name: upgrade-master | Untaint master
-  shell: "kubectl --kubeconfig=/home/{{ admin_user.name }}/.kube/config taint node {{ groups['kubernetes_master'][0] }} node-role.kubernetes.io/master:NoSchedule-"
-  become_user: "{{ admin_user.name }}"
   when:
-    - untaint_master is defined and untaint_master == True
+    - groups['kubernetes_node'] is defined
+    - groups['kubernetes_node']|length > 0 # master is drained only if there is at least one worker node
```
Comment (toszo) [inline core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml]:
```suggestion
    - groups['kubernetes_node']|list|length > 0 # master is drained only if there is at least one worker node
```

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml (hunk 1 above)
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml]: In this case casting to list is not needed (tested)

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml (hunk 2 above)
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-master.yml]: In this case casting to list is not needed (tested)

**Pull request merged**